geomstats/geometry/pullback_metric.py

```diff
@@ -126,38 +181,107 @@ def inner_product_derivative_matrix(self, base_point=None):
             Inner-product derivative matrix, where the index of the derivation
             is last: :math:`mat_{ij}_k = \partial_k g_{ij}`.
         """
-        hessian_aij = []
-        jacobian_ai = []
-        embedding_dim = self.embedding_metric.dim
-        for a in range(embedding_dim):
+        initial_ndim = base_point.ndim
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
+        inner_prod_deriv_mats = []
+        for point in base_point:
+            jacobian_ai = self.jacobian_immersion(point)
+            if self.dim == 1 and jacobian_ai.ndim > 2:
```

Comment:
TODO: address the inconsistent squeezing for function with domains of dim 1 across backends, directly in the backends.